hack/second-controller.py

```diff
@@ -0,0 +1,185 @@
+
+    parser.add_argument(
+        "--secret",
+        help="name of the secret to use for the controller, default label-secret",
+        default=os.environ.get("PAC_CONTROLLER_SECRET"),
+    )
+
+    parser.add_argument(
+        "--controller-image",
+        help="use this image for the controller, instead of the default ones, (use the keyword ko for ko)",
+        default=os.environ.get(
+            "PAC_CONTROLLER_IMAGE",
+            "ghcr.io/openshift-pipelines/pipelines-as-code-controller:stable",
+        ),
+    )
+
+    parser.add_argument(
+        "--gosmee-image",
+        help="use this image instead for gosmee",
+        default="ghcr.io/chmouel/gosmee:main",
+    )
+
+    parser.add_argument(
+        "--smee-url",
+        help="if set this will deploy a gosmee container to that smee_url and redirect query to the new controller to it",
+        default=os.environ.get("PAC_CONTROLLER_SMEE_URL"),
+    )
+
+    parser.add_argument(
+        "--namespace",
+        help="namespace where pac is installed",
+        default=os.environ.get("PAC_CONTROLLER_TARGET_NS", "pipelines-as-code"),
+    )
+
+    return parser.parse_args()
+
+
+args = parse_arguments()
+if not args.configmap:
+    args.configmap = f"{args.label}-configmap"
+
+if not args.secret:
+    args.secret = f"{args.label}-secret"
+
+controller = {}
+with open("config/400-controller.yaml", "r", encoding="utf-8") as f:
+    controller = yaml.load(f, Loader=yaml.FullLoader)
+    controller["spec"]["selector"]["matchLabels"]["app.kubernetes.io/name"] = (
+        args.label + "-controller"
+    )
+    controller["metadata"]["name"] = args.label + "-controller"
+    controller["spec"]["template"]["metadata"]["labels"]["app"] = (
+        args.label + "-controller"
+    )
+    controller["spec"]["template"]["metadata"]["labels"]["app.kubernetes.io/name"] = (
+        args.label + "-controller"
+    )
+    for container in controller["spec"]["template"]["spec"]["containers"]:
+        if container["name"] == "pac-controller":
+            if args.controller_image and args.controller_image != "ko":
+                container["image"] = args.controller_image
+            for env in container["env"]:
+                if env["name"] == "PAC_CONTROLLER_LABEL":
+                    env["value"] = args.label + "-controller"
+                if env["name"] == "PAC_CONTROLLER_SECRET":
+                    env["value"] = args.secret
+                if env["name"] == "PAC_CONTROLLER_CONFIGMAP":
+                    env["value"] = args.configmap
+
+service = {}
+with open("config/401-controller-service.yaml", "r", encoding="utf-8") as f:
+    service = yaml.load(f, Loader=yaml.FullLoader)
+    service["spec"]["selector"]["app.kubernetes.io/name"] = args.label + "-controller"
+    service["metadata"]["name"] = args.label + "-controller"
+    service["metadata"]["labels"]["app"] = args.label
+    service["metadata"]["labels"]["app.kubernetes.io/name"] = args.label + "-controller"
+
```

Comment:
It would be beneficial to also incorporate this for Openshift Route, I mean mirroring the approach used for the service.

i.e
```
route = {}
with open("config/openshift/10-routes.yaml", "r", encoding="utf-8") as f:
```

Reply:
that would makes things specific to openshift then?

Reply:
That's true 

How about determining, based on the provided parameter, whether to generate a route or not? However, this is more of an optional enhancement to make it easy for OpenShift. If you believe it might add complexity with user input, we can skip implementing it."

Reply:
i added and option in latest commit,